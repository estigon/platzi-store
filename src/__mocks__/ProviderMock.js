import React from "react";
import { createStore } from "redux";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createBrowserHistory } from "history";//se encarga se saber la estructura y a donde se ha movido nuestra aplicacion
import initialState from "../initialState";
import reducer from "../reducers";

const store = createStore(reducer, initialState);
const history = createBrowserHistory();

const ProviderMock = props => (
    <Provider store={store}>
        <Router history={history}>
            {props.children}//componentes que necesitan ser estrcuturados desde este mock
        </Router>
    </Provider>
);

export default ProviderMock;