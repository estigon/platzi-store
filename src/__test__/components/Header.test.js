import React from "react";
import { mount, shallow } from 'enzyme';
import Header from "../../components/Header";
import ProviderMock from "../../__mocks__/ProviderMock";
import {create} from 'react-test-renderer';

describe('<Header />', () => {

    test('Render del componente <Header />', () => {
        const header = shallow(
            <ProviderMock>
                <Header />
            </ProviderMock>
        );
        console.log(header.debug())
        expect(header.length).toEqual(1);
    });

    test('Render del titulo <Header />', () => {
        const header = mount(
            <ProviderMock>
                <Header />
            </ProviderMock>
        );
        console.log(header.debug())
        expect( header.find(".Header-title").text()).toEqual("Platzi Store");
    })
})

describe('Snapshot <Header />', () => {
    
    test('comprobar UI del elemento <Header />', () => {
        const header = create(
            <ProviderMock>
                <Header />
            </ProviderMock>
        )

        expect(header.toJSON()).toMatchSnapshot()
    })
})