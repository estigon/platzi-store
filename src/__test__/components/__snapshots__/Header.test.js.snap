// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot <Header /> comprobar UI del elemento <Header /> 1`] = `
Array [
  <div
    className="Header"
  >
    <h1
      className="Header-title"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Platzi Store
      </a>
    </h1>
    <div
      className="Header-checkout"
    >
      <a
        href="/checkout"
        onClick={[Function]}
      >
        <i
          className="fas fa-shopping-basket"
        />
      </a>
    </div>
  </div>,
  "//componentes que necesitan ser estrcuturados desde este mock",
]
`;
